src/Security/Security.php

```diff
@@ -570,6 +596,32 @@ protected function preLogin()
     }
 
     /**
+     * Perform pre-logout checking and prepare a response if available prior to logout
+     *
+     * @return HTTPResponse Substitute response object if the logout process should be curcumvented.
+     * Returns null if should proceed as normal.
+     */
+    protected function preLogout()
```

Comment:
I added this here so it would be consistent with `Security::login()` - that method calls `preLogin()` before delegating to the handlers. If I move the token checking to `LogoutHandler` (as per above comment), then this _could_ be removed... I think it’s still potentially useful though